test/miscellaneous_tests/symbolic_stoichiometry.jl

```diff
@@ -1,256 +1,247 @@
+
+        fake_integrator1 = (u = copy(u0_2), p = ps_2, t = τ)
+        fake_integrator2 = deepcopy(fake_integrator1)
+        catalyst_vrj.affect!(fake_integrator1)
+        jumps[i].affect!(fake_integrator2)
+        @test fake_integrator1 == fake_integrator2
+    end
 end
 
-### Simple Solving Tests on SIR Model ###
-
-# Gives an error. I can fix so there is not, but unsure what part of the code is what we actually are
-# testing for, and what we are not. Sam, if you have any opinions on what to preserve in this test or not,
-# tell me before I try to rewrite. 
-@test_broken let
-    @parameters α β γ k
-    @species S(t), I(t), R(t)
-    rxs = [Reaction(α, [S, I], [I], [1, 1], [2]),
-        Reaction(β, [I], [R], [1], [1])]
-    @named sir_ref = ReactionSystem(rxs, t)
-
-    rxs2 = [Reaction(α, [S, I], [I], [γ, 1], [k]),
-        Reaction(β, [I], [R], [γ], [γ])]
-    @named sir = ReactionSystem(rxs2, t)
-
-    @test issetequal(unknowns(sir_ref), unknowns(sir))
-
-    # ODEs.
-    p1 = (α => 0.1 / 1000, β => 0.01)
-    p2 = (α => 0.1 / 1000, β => 0.01, γ => 1, k => 2)
-    tspan = (0.0, 250.0)
-    u0 = [S => 999.0, I => 1.0, R => 0.0]
-    oprob = ODEProblem(sir_ref, u0, tspan, p1)
-    sol = solve(oprob, Tsit5())
-    # here we hack around https://github.com/SciML/ModelingToolkit.jl/issues/1475
-    p2dict = Dict(p2)
-    pvs = Tuple(p2dict[s] for s in parameters(sir))
-    @test all(isequal.(parameters(sir), parameters(convert(ODESystem, sir))))
-    oprob2 = ODEProblem(sir, u0, tspan, pvs)
-    sol2 = solve(oprob2, Tsit5())
-    @test norm(sol - sol2(sol.t)) < 1e-10
-
-    # Jumps.
-    Nsims = 10000
-    u0 = [S => 999, I => 1, R => 0]
-    jsys = convert(JumpSystem, sir_ref)
-    dprob = DiscreteProblem(jsys, u0, tspan, p1)
-    jprob = JumpProblem(jsys, dprob, Direct(); rng, save_positions = (false, false))
-    function getmean(jprob, tf)
-        m = zeros(3)
-        for i in 1:Nsims
-            sol = solve(jprob, SSAStepper())
-            m .+= sol(tspan[2])
-        end
-        m /= Nsims
-        m
+# Tests symbolic stoichiometries in simulations.
+# Tests for decimal numbered symbolic stoichiometries.
+let
+    rs_int = @reaction_network begin
+        @parameters n1::Int64 n2::Int64
+        p, 0 -->X
+        k, n1*X --> n2*Y
+        d, Y --> 0
+    end
+    rs_dec = @reaction_network begin
+        @parameters n1::Float64 n2::Float64
+        p, 0 -->X
+        k, n1*X --> n2*Y
+        d, Y --> 0
     end
-    m1 = getmean(jprob, tspan[2])
-    jsys2 = convert(JumpSystem, sir)
-    p2dict = Dict(p2)
-    pvs = Tuple(p2dict[s] for s in parameters(sir))
-    @test all(isequal.(parameters(sir), parameters(jsys2)))
-    dprob2 = DiscreteProblem(jsys2, u0, tspan, pvs)
-    jprob2 = JumpProblem(jsys2, dprob2, Direct(); rng, save_positions = (false, false))
-    m2 = getmean(jprob2, tspan[2])
-    @test maximum(abs.(m1[2:3] .- m2[2:3]) ./ m1[2:3]) < 0.05
```

Comment:
Ok, thought this was some hack you did to run the tests, more than happy to add it back in.

Reply:
Actually, you are only checking that the two systems yield the same mean trajectories, right? I thought you meant hat there was some real know statistics that you compared both systems to. Still happy to do a test that checks that they generate the same mean, in addition to identical simulations with same seed though, adding in back right now.

Reply:
Yes, it is a test that the symbolic stoich system ends up generating the same mean as the explicit stoichiometry system... They don't need the same seed since if the test passes just using the StableRNG rng it should continue passing in the future.

Reply:
Not sure if what you mean with the last sentence, but the more tests the better.

Reply:
fixed